packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -39,7 +41,23 @@ export interface FileDownloadCards {
 export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
   const truncateLength = 15;
   const { userId, message } = props;
+  const [showSpinner, setShowSpinner] = useState(false);
   const fileDownloads: FileMetadata[] = message.metadata ? extractFileMetadata(message.metadata) : [];
+  const fileDownloadHandler = useCallback(
+    async (userId, file) => {
+      if (props.downloadHandler) {
+        setShowSpinner(true);
+        const url = await props.downloadHandler(userId, file);
+        if (url instanceof URL) {
+          window.open(url.toString(), '_blank', 'noopener,noreferrer');
+        }
+        setShowSpinner(false);
+      } else {
+        window.open(file.url, '_blank', 'noopener,noreferrer');
```

Comment:
I recommend adding an if check at the top of this function to make the code easier to read.
```tsx
if (!props.downloadHandler) {
  return window.open(file.url, '_blank', 'noopener,noreferrer');
}
```

the rest of the code can then be written outside the conditional block.

Reply:
done